File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {

        int right = nums.size() / 2;
        int left = 0; 

        while(right < nums.size()){
            if(nums[left] > nums[right]){
                while(nums[right] > nums[right-1]){
                    right--;
                }
                return nums[right];
            }
            left++;
            right++;
        }

        return nums[0];
        
    }
};

File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:

    def allsubsets(self, nums: List[int], subsets: set, curr: int, curr_set: List[int]) -> None:
        if curr >= len(nums): 
            return

        for curr in range(curr, len(nums)): 
            curr_set.append(nums[curr])
            print(curr_set)
            subsets.add(tuple(curr_set))
            self.allsubsets(nums, subsets, curr+1, curr_set)
            curr_set.pop()




    def subsets(self, nums: List[int]) -> List[List[int]]:

        subsets = set()
        subsets.add(())
        curr_set = []
        response = []

        self.allsubsets(nums, subsets, 0, curr_set)

        for ind_set in subsets: 
            response.append(list(ind_set))

        return response
        